src/subhalo_tracking.cpp

```diff
@@ -1113,6 +1115,55 @@ class SubhaloMasker_t
     }
     subhalo.Particles.resize(it_save - it_begin);
   }
+
+  /* This routine masks particles by giving priority to subhaloes shallower
+   * in the hierarchy. */
+  void MaskTopBottom(HBTInt subid, vector<Subhalo_t> &Subhalos)
+  {
+    /* We perform the masking first */
+    auto &subhalo = Subhalos[subid];
+    
+    /* To keep track of how many tracers we have encountered so far. */
+    HBTInt tracer_counter = 0;
+
+    for (HBTInt i = 0; i < subhalo.Particles.size(); i++)
+    {
+      /* We have now ensured to keep the required number of most bound subset of 
+       * tracers for this subhalo. */
+      if(tracer_counter == HBTConfig.MinNumTracerPartOfSub)
+        break;
+      
+      /* Only mask tracers */
+      if(subhalo.Particles[i].IsTracer())
+      {
+        /* We insert in the exclusion list so that more deeply nested subhaloes
+         * do not remove this tracer when masking from bottom to top. */
+        auto insert_status = ExclusionList.insert(subhalo.Particles[i].Id);
+        
+        /* We should have only just one bound unique tracer within the FOF level,
+         * so we should not fail the insertion. */
+        assert(insert_status.second == true);
+
+        /* Since we do not change the size of subhalo particles at this stage, 
+         * we move forward the tracers, so we know how many to skip in the bottom
+         * to top masking step. */
+        std::swap(subhalo.Particles[i], subhalo.Particles[tracer_counter++]); 
```

Comment:
I do not think this affects any of our analysis in the following snapshot. We only need the tracers for host finding, which remain in their relative binding energy ordering. If anything, it saves a few iterations since we will always find them at the beginning of `Particles`